### PR TITLE
Add a k9s module

### PR DESCRIPTION
**Pull request by mrcjkb**

### CODE REVIEW

File: modules/k9s.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+in {
+  options.stylix.targets.k9s.enable =
+    config.lib.stylix.mkEnableTarget "k9s" true;
+
+  config = lib.mkIf config.stylix.targets.k9s.enable {
+
+    home-manager.sharedModules = [{
+      # Make sure there's no conflicting config
+      programs.k9s.skin = null;
```
Comment (danth) [inline modules/k9s.nix:21]: Would it be feasible to use this option to set the scheme?

That would avoid creating the config file when k9s is not installed.

Comment (mrcjkb) [reply modules/k9s.nix:21]: I could try using `yq` to convert the yaml to json and then call `builtins.fromJSON`. I'll have a go at it later.

Comment (mrcjkb) [reply modules/k9s.nix]: Took me a few attempts, but it works now :smile:

**Comment by danth** [comment]
NIce work, but perhaps it'd be simpler to hardcode the theme - as in the Qutebrowser module for example.

I went ahead and converted the template you are currently using:

```nix
# Based on https://github.com/krgn/base16-k9s by Karsten Gebbert

with config.lib.stylix.colors;

{
  k9s = {
    body = {
      fgColor = base05-hex;
      bgColor = "default";
      logoColor = base0C-hex;
    };

    prompt = {
      fgColor = base05-hex;
      bgColor = base00-hex;
      suggestColor = base0A-hex;
    };

    info = {
      fgColor = base0B-hex;
      sectionColor = base05-hex;
    };

    dialog = {
      fgColor = base05-hex;
      bgColor = "default";
      buttonFgColor = base05-hex;
      buttonBgColor = base0C-hex;
      buttonFocusFgColor = base0E-hex;
      buttonFocusBgColor = base0B-hex;
      labelFgColor = base0A-hex;
      fieldFgColor = base05-hex;
    };

    frame = {
      border = {
        fgColor = base02-hex;
        focusColor = base01-hex;
      };

      menu = {
        fgColor = base05-hex;
        keyColor = base0B-hex;
        numKeyColor = base0B-hex;
      };

      crumbs = {
        fgColor = base05-hex;
        bgColor = base01-hex;
        activeColor = base01-hex;
      };

      status = {
        newColor = base08-hex;
        modifyColor = base0C-hex;
        addColor = base09-hex;
        errorColor = base0D-hex;
        highlightcolor = base0A-hex;
        killColor = base03-hex;
        completedColor = base03-hex;
      };

      title = {
        fgColor = base05-hex;
        bgColor = base01-hex;
        highlightColor = base0A-hex;
        counterColor = base0C-hex;
        filterColor = base0B-hex;
      };
    };

    views = {
      charts = {
        bgColor = "default";
        defaultDialColors = [ base0C-hex base0D-hex ];
        defaultChartColors = [ base0C-hex base0D-hex ];
      };

      table = {
        fgColor = base05-hex;
        bgColor = "default";
        header = {
          fgColor = base05-hex;
          bgColor = "default";
          sorterColor = base08-hex;
        };
      };

      xray = {
        fgColor = base05-hex;
        bgColor = "default";
        cursorColor = base01-hex;
        graphicColor = base0C-hex;
        showIcons = false;
      };

      yaml = {
        keyColor = base0B-hex;
        colonColor = base0C-hex;
        valueColor = base05-hex;
      };

      logs = {
        fgColor = base05-hex;
        bgColor = "default";
        indicator = {
          fgColor = base05-hex;
          bgColor = base0C-hex;
        };
      };

      help = {
        fgColor = base05-hex;
        bgColor = base00-hex;
        indicator.fgColor = base0D-hex;
      };
    };
  };
}
```

The original template is under the MIT license so we are free to do this.

Could you combine this with your module and check whether it works?

**Comment by mrcjkb** [comment]
>  Could you combine this with your module and check whether it works?

I can try it out. Would you like me to remove the yaml2attrs helper functions then?

**Comment by mrcjkb** [comment]
> Could you combine this with your module and check whether it works?

It evaluates and builds, but it seems like I will have to modify the values a bit. The current implementation doesn't work properly with k9s.

```diff
1,7d0
< background: '#212121'
< blue: '#c3e88d'
< comment: '#4a4a4a'
< current_line: '#303030'
< cyan: '#f07178'
< foreground: '#eeffff'
< green: '#f78c6c'
11,12c4,5
<     fgColor: '#eeffff'
<     logoColor: '#89ddff'
---
>     fgColor: eeffff
>     logoColor: 89ddff
15,21c8,14
<     buttonBgColor: '#89ddff'
<     buttonFgColor: '#eeffff'
<     buttonFocusBgColor: '#c3e88d'
<     buttonFocusFgColor: '#c792ea'
<     fgColor: '#eeffff'
<     fieldFgColor: '#eeffff'
<     labelFgColor: '#ffcb6b'
---
>     buttonBgColor: 89ddff
>     buttonFgColor: eeffff
>     buttonFocusBgColor: c3e88d
>     buttonFocusFgColor: c792ea
>     fgColor: eeffff
>     fieldFgColor: eeffff
>     labelFgColor: ffcb6b
24,25c17,18
<       fgColor: '#353535'
<       focusColor: '#303030'
---
>       fgColor: '353535'
>       focusColor: '303030'
27,29c20,22
<       activeColor: '#303030'
<       bgColor: '#303030'
<       fgColor: '#eeffff'
---
>       activeColor: '303030'
>       bgColor: '303030'
>       fgColor: eeffff
31,33c24,26
<       fgColor: '#eeffff'
<       keyColor: '#c3e88d'
<       numKeyColor: '#c3e88d'
---
>       fgColor: eeffff
>       keyColor: c3e88d
>       numKeyColor: c3e88d
35,41c28,34
<       addColor: '#f78c6c'
<       completedColor: '#4a4a4a'
<       errorColor: '#82aaff'
<       highlightcolor: '#ffcb6b'
<       killColor: '#4a4a4a'
<       modifyColor: '#89ddff'
<       newColor: '#f07178'
---
>       addColor: f78c6c
>       completedColor: 4a4a4a
>       errorColor: 82aaff
>       highlightcolor: ffcb6b
>       killColor: 4a4a4a
>       modifyColor: 89ddff
>       newColor: f07178
43,47c36,40
<       bgColor: '#303030'
<       counterColor: '#89ddff'
<       fgColor: '#eeffff'
<       filterColor: '#c3e88d'
<       highlightColor: '#ffcb6b'
---
>       bgColor: '303030'
>       counterColor: 89ddff
>       fgColor: eeffff
>       filterColor: c3e88d
>       highlightColor: ffcb6b
49,50c42,43
<     fgColor: '#c3e88d'
<     sectionColor: '#eeffff'
---
>     fgColor: c3e88d
>     sectionColor: eeffff
52,54c45,47
<     bgColor: '#212121'
<     fgColor: '#eeffff'
<     suggestColor: '#ffcb6b'
---
>     bgColor: '212121'
>     fgColor: eeffff
>     suggestColor: ffcb6b
59,60c52,53
<       - '#89ddff'
<       - '#82aaff'
---
>       - 89ddff
>       - 82aaff
62,63c55,56
<       - '#89ddff'
<       - '#82aaff'
---
>       - 89ddff
>       - 82aaff
65,66c58,59
<       bgColor: '#212121'
<       fgColor: '#eeffff'
---
>       bgColor: '212121'
>       fgColor: eeffff
68c61
<         fgColor: '#82aaff'
---
>         fgColor: 82aaff
71c64
<       fgColor: '#eeffff'
---
>       fgColor: eeffff
73,74c66,67
<         bgColor: '#89ddff'
<         fgColor: '#eeffff'
---
>         bgColor: 89ddff
>         fgColor: eeffff
77c70
<       fgColor: '#eeffff'
---
>       fgColor: eeffff
80,81c73,74
<         fgColor: '#eeffff'
<         sorterColor: '#f07178'
---
>         fgColor: eeffff
>         sorterColor: f07178
84,86c77,79
<       cursorColor: '#303030'
<       fgColor: '#eeffff'
<       graphicColor: '#89ddff'
---
>       cursorColor: '303030'
>       fgColor: eeffff
>       graphicColor: 89ddff
89,96c82,84
<       colonColor: '#89ddff'
<       keyColor: '#c3e88d'
<       valueColor: '#eeffff'
< magenta: '#89ddff'
< orange: '#ffcb6b'
< red: '#82aaff'
< selection: '#353535'
< yellow: '#c792ea'
---
>       colonColor: 89ddff
>       keyColor: c3e88d
>       valueColor: eeffff
```

**Comment by mrcjkb** [comment]
Now it works!

**Review by danth — APPROVED** [review]
Great, thanks!

**Pull request merged**